test/user_defined_output/fake_plugin/fake_user_defined_output.cc

```diff
@@ -69,28 +77,33 @@ FakeUserDefinedOutputPluginFactory::createUserDefinedOutputPlugin(
 }
 
 absl::StatusOr<Envoy::ProtobufWkt::Any> FakeUserDefinedOutputPluginFactory::AggregateGlobalOutput(
-    absl::Span<const Envoy::ProtobufWkt::Any> per_worker_outputs) {
+    absl::Span<const nighthawk::client::UserDefinedOutput> per_worker_outputs) {
   FakeUserDefinedOutput global_output;
   global_output.set_worker_name("global");
   int data_called = 0;
   int headers_called = 0;
-  for (const Envoy::ProtobufWkt::Any& any : per_worker_outputs) {
-    FakeUserDefinedOutput output;
-    absl::Status status = Envoy::MessageUtil::unpackToNoThrow(any, output);
-    if (status.ok()) {
-      data_called += output.data_called();
-      headers_called += output.headers_called();
+  for (const nighthawk::client::UserDefinedOutput& user_defined_output : per_worker_outputs) {
+    if (user_defined_output.has_typed_output()) {
+      Envoy::ProtobufWkt::Any any = user_defined_output.typed_output();
+      FakeUserDefinedOutput output;
+      absl::Status status = Envoy::MessageUtil::unpackToNoThrow(any, output);
+      if (status.ok()) {
+        data_called += output.data_called();
+        headers_called += output.headers_called();
+      } else {
+        return status;
+      }
     } else {
-      return status;
+      return absl::InvalidArgumentError(
```

Comment:
Is this what we expect the real plugin to do? Or should it aggregate the error messages and return the output to make some results from the run useful?

Related - since we need to aggregate partial errors into the global error - should the proto carry a list strings (errors) instead of just a single error?